ochap-authentication/src/main/java/com/example/ochapauthentication/routes/AuthenticationController.java

```diff
@@ -0,0 +1,141 @@
+public class AuthenticationController {
+
+    final UserDAO userRepository;
+    final RoleDAO roleRepository;
+    final ObjectMapper objectMapper;
+
+    final int MIN_USERNAME_LENGTH = 3;
+    final int MIN_PASSWORD_LENGTH = 8;
+
+    @Autowired
+    AuthenticationController(UserDAO userRepository, RoleDAO roleRepository){
+        this.userRepository = userRepository;
+        this.roleRepository = roleRepository;
+        this.objectMapper = new ObjectMapper();
+    }
+
+    @Autowired
+    private JwtTokenUtil jwt;
+
+    @PostMapping("/auth/login")
+    @ResponseBody
+    ResponseEntity<String> loginAutentication(@RequestBody AuthLoginCommand credentials) throws JsonProcessingException {
+        Optional<User> user = userRepository.getUserByUsername(credentials.getUsername());
+
+        if (user.isEmpty() || !Arrays.equals(hashPassword(credentials.getPassword(), user.get().getSalt()),
+                                             user.get().getPassword())) {
+            com.example.ochapauthentication.dto.Error incorrectCredentials =
+                    new com.example.ochapauthentication.dto.Error("Incorrect credentials");
+
+            return new ResponseEntity<>(objectMapper.writeValueAsString(incorrectCredentials), HttpStatus.FORBIDDEN);
+        }
+
+        // Generate JWT
+        User u = user.get();
+        AccountInfoDTO accountInfo = new AccountInfoDTO(u.getId(), u.getUsername(), u.getRole().getName());
+
+        TokenDTO tokenDTO = new TokenDTO(jwt.generateToken(accountInfo), accountInfo);
+
+        return new ResponseEntity<>(objectMapper.writeValueAsString(tokenDTO), HttpStatus.OK);
+    }
+
+    @PostMapping(value = "/accounts/register",  produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    ResponseEntity<String> registerAutentication(@RequestBody AccountRegisterCommand credentials) throws JsonProcessingException {
+
+        boolean isError = false;
+        ArrayList<ErrorDTO> errorList = new ArrayList<>();
+        if(credentials.getUsername() == null || credentials.getUsername().length() < MIN_USERNAME_LENGTH) {
+            isError = true;
+            errorList.add(new ErrorDTO("username", "Must be at least " + MIN_USERNAME_LENGTH + " characters long"));
+        }
+
+        if(credentials.getPassword() == null || credentials.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            isError = true;
+            errorList.add(new ErrorDTO("password", "Must be at least " + MIN_PASSWORD_LENGTH + " characters long"));
+        }
+
+        if(isError){
+            ErrorsDTO errors = new ErrorsDTO(errorList);
+            return new ResponseEntity<>(objectMapper.writeValueAsString(errors), HttpStatus.UNPROCESSABLE_ENTITY);
+        }
+
+        if (userRepository.getUserByUsername(credentials.getUsername()).isPresent()) {
+            com.example.ochapauthentication.dto.Error alreadyExisting =
+            new com.example.ochapauthentication.dto.Error("The username already exist");
```

Comment:
```suggestion
            new com.example.ochapauthentication.dto.Error("A user with the specified username already exists.");
```

The username identifies a user, but is not an entity on its own.